utils/prof_utils.py

```diff
@@ -0,0 +1,1150 @@
+        alphas.append(alpha_key)
+    best_chi = min(chi_diff)
+    best_alpha = alphas[chi_diff.index(best_chi)]
+    fit_dict = attempts_dict[best_alpha]
+
+    if plot_name:
+        plot_fit(plot_name, fit_dict["profile"], fit_dict["fit"], fit_dict["gaussian_params"], maxima=fit_dict["maxima_e"], maxima_e=fit_dict["maxima"])
+
+    logger.info("### Best fit results ###")
+    logger.info("Best model found with BIC of {0} and reduced Chi of {1} using an alpha value of {2}"\
+                .format(fit_dict["bic"], fit_dict["redchisq"], best_alpha))
+    logger.info("W10:                   {0} +/- {1}".format(fit_dict["W10"], fit_dict["W10_e"]))
+    logger.info("W50:                   {0} +/- {1}".format(fit_dict["W50"], fit_dict["W50_e"]))
+    logger.info("Wscat:                 {0} +/- {1}".format(fit_dict["Wscat"], fit_dict["Wscat_e"]))
+    logger.info("Weq:                   {0} +/- {1}".format(fit_dict["Weq"], fit_dict["Weq_e"]))
+    logger.info("Maxima:                {0}".format(fit_dict["maxima"]))
+    logger.info("Maxima error:          {0}".format(fit_dict["maxima_e"]))
+    if fit_dict["sn"]:
+        logger.info("S/N estimate:          {0} +/- {1}".format(fit_dict["sn"], fit_dict["sn_e"]))
+
+    return fit_dict
+
+#---------------------------------------------------------------
+if __name__ == '__main__':
+
+    loglevels = dict(DEBUG=logging.DEBUG,\
+                    INFO=logging.INFO,\
+                    WARNING=logging.WARNING,\
+                    ERROR=logging.ERROR)
+
+    parser = argparse.ArgumentParser(description="""A utility file for calculating a variety of pulse profile properties""",\
+                                    formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    inputs = parser.add_argument_group("Inputs")
+    inputs.add_argument("--bestprof", type=str, help="The pathname of the file containing the pulse profile. Use if in .pfd.bestprof format")
+    inputs.add_argument("--ascii", type=str, help="The pathname of the file containing the pulse profile. Use if in ascii text format")
+    inputs.add_argument("--ignore_threshold", type=float, default=0.02, help="Maxima with values below this fraction of the profile maximum will be ignored.")
+    inputs.add_argument("--min_comp_len", type=int, default=None,\
+                        help="Minimum length of a component to be considered real. Measured in bins. If none, will use 1 percent of total profile length")
+    inputs.add_argument("--alpha", type=float, default=2, help="Used by the clipping function to determine the noise level. A lower value indicates\
+                        a higher verbosity level in the noise clipping function.")
+    inputs.add_argument("--auto", action="store_true", help="Used to automatically find the best alpha value to clip this profile")
+    inputs.add_argument("--period", type=float, help="The period of the puslar in ms. Found automatically if .bestprof supplied. Used in S/N calculation. Not required")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [line too long (168 > 159 characters)](https://app.codacy.com/manual/CIRA-Pulsars-and-Transients-Group/vcstools/pullRequest?prid=4645369)